addons/website_forum/controllers/website_forum.py

```diff
@@ -35,7 +36,20 @@ def _prepare_user_values(self, **kwargs):
         if kwargs.get('forum'):
             values['forum'] = kwargs.get('forum')
         elif kwargs.get('forum_id'):
-            values['forum'] = request.env['forum.forum'].browse(int(kwargs.pop('forum_id')))
+            values['forum'] = Forum.browse(int(kwargs.pop('forum_id')))
+        forum = values.get('forum')
+        if isinstance(forum, Forum.__class__):
+            other_website_forums = Forum.search(request.website.website_domain()) - forum
```

Comment:
Can we use `_search` to not actually run the search at this point? That is, if we can use it in `read_group`, which I'm not sure.